src/overlays/actors/ovl_Door_Shutter/z_door_shutter.c

```diff
@@ -349,39 +357,43 @@ s32 DoorShutter_SetupDoor(DoorShutter* this, PlayState* play) {
 
     if (doorType != SHUTTER_KEY_LOCKED) {
         if (frontRoom == transitionEntry->sides[1].room) {
+            // if player is in front
             if (ABS((s16)(this->dyna.actor.shape.rot.y - this->dyna.actor.yawTowardsPlayer)) < 0x4000) {
                 frontRoom = -1;
             }
         }
         if (frontRoom == this->dyna.actor.room) {
             if (doorType == SHUTTER_FRONT_SWITCH_BACK_CLEAR) { // Swap the back clear to the front clear
                 doorType = SHUTTER_FRONT_CLEAR;
+            } else if (doorType == SHUTTER_BOSS) {
+                doorType = SHUTTER_BACK_LOCKED;
             } else {
-                doorType = (doorType == SHUTTER_BOSS) ? SHUTTER_BACK_LOCKED : SHUTTER;
+                doorType = SHUTTER;
             }
         }
     }
     this->gfxType = (doorType == SHUTTER) ? styleInfo->gfxType1 : styleInfo->gfxType2;
```

Comment:
Wasn't you, but what does `SHUTTER` by itself mean? Also could possibly give more descriptive names than `type1` and `type2` based on this ternary?

Reply:
Afaict it's just "normal" but I'm honestly not sure.

The block of code just above with `frontRoom` I'm not sure I understand either. I'm waiting on documentation for transition actors to be merged to investigate room loading and this transition actor "side" usage